backbone.nativeajax.js

```diff
@@ -44,16 +45,33 @@
 
         var status = xhr.status;
         var data = getData(options.headers && options.headers.Accept, xhr);
+        var aCallbackFailed = false;
+        var reason;
 
         // Check for validity.
         if (isValid(xhr)) {
-          if (options.success) options.success(data);
-          if (resolve) resolve(data);
+          try {
+            options.success && options.success(data);
+          } catch(e) {
+            reason = e;
```

Comment:
I believe, though I haven't tested, that there shouldn't be a need for the `try..catch`. It should all get caught in the promise `catch` handler.